cmd/operator/main.go

```diff
@@ -303,16 +304,14 @@ func Main() int {
 		}
 	}
 
-	// todo - I wonder can these go away because of
-	// https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#admission-webhook-metrics
 	validationTriggeredCounter := prometheus.NewCounter(prometheus.CounterOpts{
 		Name: "prometheus_operator_rule_validation_triggered_total",
 		Help: "Number of times a prometheusRule object triggered validation",
 	})
 
 	validationErrorsCounter := prometheus.NewCounter(prometheus.CounterOpts{
 		Name: "prometheus_operator_rule_validation_errors_total",
-		Help: "Number of errors that occurred while validating a prometheusRules object",
+		Help: "DEPRECATED: Number of errors that occurred while validating a prometheusRules object",
```

Comment:
Can we say here in which operator version this metric won't be available anymore?

Reply:
I spoke to @simonpasquier about this before and I think the idea was that we give it too releases and remove. However  I think we have marked things as deprecated and never really removed them so some clarity around that process would be beneficial.

Reply:
AFAIU the deprecation policy depends on what is modified:
* For CRD, we apply https://kubernetes.io/docs/reference/using-api/deprecation-policy/. In practice, we can't remove fields from a v1 CRD until an equivalent v2 CRD exists.
* Operator flags, metrics and behavior (to some extent) can be changed after a deprecation period which is usually 2 releases.

I might be good to agree on a common definition and document it in CONTRIBUTING.md?

Reply:
I think it's fine to follow the 2 releases deprecation period.

```suggestion
		Help: "Number of errors that occurred while validating a prometheusRules object (DEPRECATED, removed in v0.57.0)",
```

Reply:
looks good. Was going to add it to the commit message but this is more obvious